src/bin/genemichaels.rs

```diff
@@ -82,14 +89,23 @@ impl FromStr for On {
     }
 }
 
-#[derive(Parser)]
+#[derive(Parser, Clone)]
 struct Args {
     #[arg(help = "Files to format in place; if none specified formats stdin and writes result to stdout")]
     files: Vec<PathBuf>,
-    #[arg(short, long, default_value_t = FormatConfig::default().max_width)]
-    line_length: usize,
     #[arg(short, long, help = "Won't emit any output")]
     quiet: bool,
+    #[arg(
+        short,
+        long,
+        help =
+            "Formats the entire workspace\n  - if your project is single folder, put an empty [workspace] in your Cargo.toml",
+    )]
+    workspace: bool,
```

Comment:
I'm wondering if it would be better to rename this `cargo` or `project` or `auto` or something, since I don't think there's a reason to limit it explicitly to workspaces.

Reply:
I think this is now solved with the new format - I'm unconvinced on the name change though.

Reply:
Sorry, thinking about this more I feel like limiting this to _just_ workspaces complicates the usage too much.  Is there some reason you'd want to format code but only if it's in any workspace, rather than the whole project?  I think the main reason for this feature is to exclude `target` which doesn't necessitate just workspaces.

Reply:
I think I see where you're going - ok fair enough, considering the new flow of parsing any old Cargo.toml I think it *is* reasonable to rename this to project.